### PR TITLE
add a basic typescript impl

### PR DESCRIPTION
**Pull request by edjroz**

### CODE REVIEW

File: spec/client/ics-008-wasm-client/README.md [hunk 1]
```diff
@@ -271,14 +270,14 @@ function upgradeClientState(
 Wasm client state verification functions check a Merkle proof against a previously validated commitment root.
 
 ```typescript
-	function () VerifyUpgradeAndUpdateState(
+	function VerifyUpgradeAndUpdateState(
     c: ClientState,
```
Comment (blasrodri) [inline spec/client/ics-008-wasm-client/README.md]: formatting here

File: spec/client/ics-008-wasm-client/README.md [hunk 2]
```diff
@@ -327,30 +326,33 @@ func (c ClientState) VerifyClientMessage(ctx sdk.Context, cdc codec.BinaryCodec,
 	return err
 }
 
-func (c ClientState) CheckForMisbehaviour(ctx sdk.Context, cdc codec.BinaryCodec, clientStore sdk.KVStore, msg exported.ClientMessage) bool {
-	wasmMisbehaviour, ok := msg.(*Misbehaviour)
+func  CheckForMisbehaviour(ctx: sdk.Context, cdc: codec.BinaryCodec, clientStore: sdk.KVStore, msg: exported.ClientMessage): bool {
+	let [ wasmMisbehaviour, ok ] = msg.(*Misbehaviour)
```
Comment (blasrodri) [inline spec/client/ics-008-wasm-client/README.md]: is this even possible in typescript?

Comment (edjroz) [reply spec/client/ics-008-wasm-client/README.md]: It is possible to deconstruct in typescript, Ill replace the pointer reference however

```suggestion
	let { wasmMisbehaviour, ok } = msg.Misbehaviour
```

File: spec/client/ics-008-wasm-client/README.md [hunk 3]
```diff
@@ -271,86 +270,89 @@ function upgradeClientState(
...
-    assert(err == null)
-    return null
+	function VerifyUpgradeAndUpdateState(
+        c: ClientState,
+        store: KVStore,
+        newClient: ClientState,
+        newConsState: ConsensusState,
+        proofUpgradeClient: []byte,
+        proofUpgradeConsState: []byte,
+	): error {
```
Comment (blasrodri) [inline spec/client/ics-008-wasm-client/README.md:280]: Does error makes sense in typescript?

**Pull request merged**